src/etc/inc/system.inc

```diff
@@ -1365,6 +1343,49 @@ function system_webgui_start() {
 	return $res;
 }
 
+function get_dns_nameservers() {
+	global $config;
+	
+	$dns_nameserver_number = 0;
+	$dns_nameservers = array();
+	
+	if (isset($config['system']['developerspew'])) {
+		$mt = microtime();
+		echo "get_dns_nameservers() being called $mt\n";
+	}
+	
+	$syscfg = $config['system'];
+	if ((((isset($config['dnsmasq']['enable'])) &&
+   		(empty($config['dnsmasq']['port']) || $config['dnsmasq']['port'] == "53") &&
+	      	(empty($config['dnsmasq']['interface']) ||
+	       	in_array("lo0", explode(",", $config['dnsmasq']['interface'])))) ||
+	     	((isset($config['unbound']['enable'])) &&
+	      	(empty($config['unbound']['port']) || $config['unbound']['port'] == "53") &&
+	      	(empty($config['unbound']['active_interface']) ||
+	       	in_array("lo0", explode(",", $config['unbound']['active_interface'])) ||
+	       	in_array("all", explode(",", $config['unbound']['active_interface']), true)))) &&
+	     	(!isset($config['system']['dnslocalhost']))) {
+			$dns_nameservers[$i++] = "127.0.0.1";
+	}
+	if (isset($syscfg['dnsallowoverride'])) {
+	/* get dynamically assigned DNS servers (if any) */
+		$ns = array_unique(get_nameservers());
+		foreach ($ns as $nameserver) {
+			if ($nameserver) {
+				$dns_nameservers[$i++] = "$nameserver";
+			}
+		}
+	}
+	if (is_array($syscfg['dnsserver'])) {
+		foreach ($syscfg['dnsserver'] as $sys_dnsserver) {
+			if ($sys_dnsserver && (!in_array($sys_dnsserver, $ns))) {
+				$dns_nameservers[$i++] = "$sys_dnsserver";
+			}
+		}
+	}
+	return $dns_nameservers;
```

Comment:
As long as we're collecting these in an array, for good measure this should probably do `return array_unique($dns_nameservers);`, to avoid using the same server more than once.

Reply:
good idea. changed